Formula/swift.rb

```diff
@@ -46,144 +46,159 @@ class Swift < Formula
 
   on_linux do
     depends_on "six" => :build
+    depends_on "icu4c" # Used in swift-corelibs-foundation
```

Comment:
Still? I thought they ditched ICU.

Reply:
[It was indeed ditched from the Swift stdlib](https://github.com/apple/swift/commit/21ee3a5e0f8cdc33232d7bccf44df26116946896), but [it still exists in corelibs-foundation](https://github.com/apple/swift-corelibs-foundation/blob/swift-5.6-RELEASE/CMakeLists.txt#L50).

I moved the dep to `on_linux` since "uses from macOS" is no longer semantically true after the removal from the stdlib - the remaining part in corelibs-foundation is not built at all on macOS.